normal/matrix.py

```diff
@@ -0,0 +1,107 @@
+def __get_cols_number(matrix):
+    return len(matrix[0])
+
+
+def __get_rows_number(matrix):
+    return len(matrix)
+
+
+def __get_nth_row(matrix, n):
+    return matrix[n]
+
+
+def __get_nth_col(matrix, n):
+    col = []
+    for row in matrix:
+        col.append(row[n])
+    return col
+
+
+def __init_result_matrix_of_size(rows_num, cols_num):
+    matrix = []
+    for row_num in range(rows_num):
+        matrix.append([])
+        for col_num in range(cols_num):
+            matrix[row_num].append(0)
+    return matrix
+
+
+def __compute_sum_of_multiplications(vector1, vector2):
+    sum = 0
+    for index, value in enumerate(vector1):
+        sum += value * vector2[index]
+    return sum
+
+
+def __validate_sizes(matrix_1, matrix_2):
+    if __get_cols_number(matrix_1) != __get_rows_number(matrix_2):
+        raise ValueError('Invalid sizes of matrices')
+
+
+def __validate_if_matrix_empty(matrix):
+    if len(matrix) == 0:
+        raise ValueError("Empty matrix")
+
+
+def __validate_matrix_first_empty_row(matrix):
+    if len(matrix[0]) == 0:
+        raise ValueError("First empty row of matrix")
+
+
+def __validate_rows_length_equality(matrix):
+    first_row_len = len(matrix[0])
+    for row in matrix:
+        if len(row) != first_row_len:
+            raise ValueError("Different sizes of matrix rows")
```

Comment:
Maybe use custom error class implementation

Reply:
I think ValueError is acceptable